### PR TITLE
Feedback pls 

### PR DESCRIPTION
**Pull request by natbibi**

### CODE REVIEW

File: api/app.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const express = require('express');
```
Comment (gretaivan) [inline api/app.js:1]: README file would be useful at least to have commands how to run

File: api/controllers/posts.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const express = require('express');
+const router = express.Router();
+
+const Post = require('../model/dog')
+
+// all posts route //
+router.get('/', async(req, res) => {
+    try {
+        const posts = await Post.all
+        res.json({posts})
```
Comment (gretaivan) [inline api/controllers/posts.js:10]: could have success status

File: api/dbConfig/init.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const { Pool } = require("pg");
+
+const pool = new Pool({ database: process.env.PGDATABASE});
```
Comment (gretaivan) [inline api/dbConfig/init.js:3]: database: process.env.... is redundant you defined your env in your docker compose and don't have dependency to handle it

File: docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+services:
+  server:
+    image: node:12.18.4
+    working_dir: /code
+    ports: 
+      - 3000:3000
+    environment: 
+      - PGUSER=futureproof
+      - PGHOST=db
+      - PGPASSWORD=shelterpass
```
Comment (natbibi) [inline docker-compose.yaml:11]: shelterpass lolol

File: requests.rest [hunk 5]
```diff
@@ -0,0 +1,12 @@
+GET http://localhost:8000/
```
Comment (natbibi) [inline requests.rest:1]: my man kyle

File: client/index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+btn.addEventListener("click", e => {
+    
+    console.log(e.target)
+    e.preventDefault();
+    console.log('button clicked')
+    const title = document.querySelector('#title').value
+    const name = document.querySelector('#name').value
+    const body = document.querySelector('#body').value
+    submitPost(title,name,body)
+    console.log('submitted post')
```
Comment (natbibi) [inline client/index.js:13]: niceeeee

File: api/server.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const cors = require('cors');
+const express = require('express');
+const app = express()
+const port = 8000
```
Comment (natbibi) [inline api/server.js:4]: you wrote Port 3000 in app.js!

File: api/dbConfig/seed.sql [hunk 8]
```diff
@@ -0,0 +1,7 @@
+INSERT INTO posts (title,name,body)
+VALUES
+(
```
Comment (gretaivan) [inline api/dbConfig/seed.sql:3]: you would no need to do this extra configuration if you had js file using fs sync reading the seed.sql file, but interesting approach

File: api/models/post.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+                let postData = await db.query(`SELECT * FROM posts WHERE id = $1;`, [ id ]);
+                let post = new Post(postData.rows[0]);
+                resolve (post);
+            } catch (err) {
+                reject('Post not found');
+            }
+        });
+    }
+}
+
```
Comment (gretaivan) [inline api/models/post.js:47]: looks great

**Review by natbibi — COMMENTED** [review]
Great start guysssss, more comments from Greta coming your way!

File: api/server.js (hunk 7 above)
Comment (gretaivan) [inline api/server.js:4]: port standard is 3000 not 8000
to make sure that server works when deployed better use : 

const port = process.env.PORT || 3000;

Comment (gretaivan) [inline api/server.js:4]: server by default should be using port 3000 it is better practice. If you want it to work post deployment you should use below: 

const port = process.env.PORT || 3000;

if you look at your app.js the comments are there.  this file should provide the implementation for server functionality and give it to the app.

File: api/server.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    res.send("Welcome to the homepage!")
+})
+
+app.post("/", (req, res) => {
+    res.send("Post worked!")
+})
+
+app.listen(port, () => {
+    console.log(`Listening at port ${port}`)
+})
```
Comment (gretaivan) [inline api/server.js:18]: You are also confusing server and app. Your entry point is an app and should be listening for request and have the API route connectors. Better use one file if you are unsure how to separate this.

File: api/app.js [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+server.use(express.json());
+
+const postRoutes = require('./controllers/posts')
+
+server.use('/posts', postRoutes)
+
+const port = process.env.PORT || 3000;
+
+// Root route
+//server.get('/', (req, res) => res.send('Hello, world!')) - might not be needed?
```
Comment (gretaivan) [inline api/app.js:16]: To fi x this you should have a server listening to requests here. You have a port but it is unused. 

so here app.listen(port,()=> {console.log("...port no"... ))})

no need to export this file as it si your main entry point actually here you import your server functionality

File: client/index.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+  };
+  console.log(postData)//working up to here
+  const options = { 
+      method: 'POST',
+      body: JSON.stringify(postData),
+      headers: { "Content-Type": "application/json" }
+  };
+  console.log(options)
+
+  fetch('http://localhost:8000/posts', options)
```
Comment (gretaivan) [inline client/index.js:32]: you are confusing your ports. Potentially your port 3000 did not work as you had unused port hanging

File: client/styles.css [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+  align-items: center;
+  justify-content: center;
+}
+
+form {
+  margin: 10px 10px 10px 10px;
+}
+
+#publish {
+  margin: 10px 10px 10px 10px;
```
Comment (gretaivan) [inline client/styles.css:12]: Well done getting some styling done! Nice minimalistic approach

**Review by gretaivan — COMMENTED** [review]
Great stuff guys!" I hope comments help to resolve your server issues and connecting both client with the server. 
Good project structure, neat code.
Nice indentation, the code is easy to read.

I liked the creative approach to get database working. 

I am sure with a little bit more time you can get this MVP up!

**Pull request closed without merging**